### ISSUE TITLE
CozyTouch is not working anymore due to login API changes [DO NOT UPDATE YOUR COZYTOUCH APP TO 3.0]

### ISSUE THREAD

**Issue by Quentame** [iMicknl/ha-tahoma#589]
### Did you read the instructions?

- [X] I have read the [README](https://github.com/iMicknl/ha-tahoma/blob/master/README.md), including the [advanced debugging](https://github.com/iMicknl/ha-tahoma/blob/master/README.md#advanced) section.

### The problem

After updating the app to 3.0.0, I opened it and then some kind of update proccess are beeing made to my account (maybe also the bridge I guess).

Here is the app release note (sorry in French even when trying to for the language to Englich):
- iOS: https://apps.apple.com/fr/app/thermor-cozytouch/id1311321675?l=en
- Android: https://play.google.com/store/apps/details?id=fr.thermor.cozytouch&hl=en (not updated to 3.0 at this time)

### What version of this integration (ha-tahoma) has the issue?

2.8.0

### What version of Home Assistant Core has the issue?

core-2021.9.6

### Which gateway / hub do you use?

Cozytouch

### Device model

unknown

### Home Assistant log

```shell
Logger: custom_components.tahoma
Source: helpers/update_coordinator.py:227
Integration: Overkiz (by Somfy)
First occurred: 10:36:25 (1 occurrences)
Last logged: 10:36:25

Authentication failed while fetching device events data: Bad credentials

I did not get better logs after setting log level to debug
```

### Additional information

Before the installation of the update, the hub firmware version was 2021.3.4

**Comment by iMicknl** [comment]
> **What's new**
> 
> Your customer account evolves, it becomes simpler and more convenient!
> Benefit from a single identifier between our different services.
> 
> Your e-mail address becomes your unique identifier for Cozytouch and your customer account.
> For new users, account creation has been simplified. You no longer need to add a Cozytouch bridge to create an account.
> 
> See you soon in Cozytouch!

Thanks for reporting, @Quentame! I can imagine that this is similar to what Nexity is doing, where we were able to overcome this limitation, but it required some work... We will investigate.

**Comment by steve41200** [comment]
I try for the first time your integration for Rexe (Econnect2)l, and he won't connect. I have login or password incorrect. I don't know if it is due to the same problem.
I have a new heater IO (I have plug a zigbeeswitch behind the other), but him is in a bathroom and I will give a shot to your integration.  I will try again and give you return of my try

**Comment by steve41200** [comment]
I have an old VM for trying a long time ago with Homebridge on it (and plug in Tahoma and it will always works to control Rexel Box via HA, the login and pass are good. The plug in is not update since long time)
I have try to a long time ago via Node Red with Overkill node directly, but I don't found them, I've erase them I think...

**Comment by iMicknl** [comment]
@steve41200 best would be to create a new issue, since the CozyTouch situation is a bit different. Happy to help you in a new issue. I think you are the first Rexel user to test this out.

**Comment by jugla** [comment]
Hello,
I have update the cozytouch app to version 3.0.0 .
Same as @Quentame and  @Hankey71 , the ha-tahoma integration no longer works :-(

Thanks in advance for your work !

**Comment by iMicknl** [comment]
I haven't had time yet to dive in to it. You can help by trying to capture the requests they do on login (e.g. with Proxyman or another proxy/MITM tool).

You can join our Discord as well, to work all together to tackle this ;-).

**Comment by iMicknl** [comment]
It seems Android has been updated as well, I will try to add support for new CozyTouch this week.

**Comment by jugla** [comment]
Hello,
no error at all during this day :-)

**Comment by Quentame** [comment]
> @hewittd we agree that I had to put all files present in ha-tahoma-feature-support_custom_authentication folder in custom_components/tahoma ? Without erase all files ?

Nope, as said here https://github.com/iMicknl/ha-tahoma/issues/589#issuecomment-940495377

Here are the steps:
1. remove your current `[ha_dir]/custom_components/tahoma` folder
2. unzip `ha-tahoma-feature-support_custom_authentication.zip`
3. copy  the `ha-tahoma-feature-support_custom_authentication/custom_components/tahoma` folder
4. paste in `[ha_dir]/custom_components` folder

**Comment by gtblaurent** [comment]
bonjour, je n'arrive pas a faire fonctionné mon cozytouch avec cette méthode je pense que je dois mal faire une manipulations j'ai remplacé le dossier tahoma par le tahoma dans l'extraction mais j'ai toujours email ou mot de passe incorrect le cozytouch et bien en version 3.0

**Comment by iMicknl** [comment]
@gtblaurent please post in English, since not everyone here speaks French. (You can use a translate engine if you want).

Thanks for sharing your stacktrace @pod31, we will have a look. Currently the error handling isn't great yet, that's why it isn't released yet.

**Comment by gtblaurent** [comment]
@iMicknl 
hello, I can't get my cozytouch to work with this method I think I must do something wrong I replaced the tahoma folder with the tahoma in the extraction but I still have the wrong email or password on the cozytouch and good in version 3.0

**Comment by browetd** [comment]
@gtblaurent did you follow the steps described in the post from @Quentame (see previous post).

**Comment by gtblaurent** [comment]
@browetd , 
yes, I followed the steps after I put back the tahoma folder which is in the archive in "custom_components"

**Comment by gtblaurent** [comment]
@iMicknl  mail send

**Comment by pod31** [comment]
How can we change the password until the cozytouch app is lock ? I'm not sure about my password 😬
